Feat(workflow): add apply components and traits docs (#158)
oam-dev/kubevela.io: docs/end-user/workflow/apply-component.md

    traits:
    - type: ingress
      properties:
        domain: testsvc.example.com
        http:
          /: 8000
  workflow:
    steps:
      - name: express-server
        type: apply-express
        properties:
          name: express-server
```

As you can see, we have a `step` with `type` of `apply-express` in `workflow`, and pass a parameter called `name` with the value of the name of the component in the application.

Next, we can write `WorkflowStepDefinition`. KubeVela provides an integrated operator, `ApplyComponent`, through which components and traits can be easily applied.

```yaml
apiVersion: core.oam.dev/v1beta1
kind: WorkflowStepDefinition
metadata:
  name: apply-express
  namespace: default
spec:
  schematic:
    cue:
      template: |
        import (
        	"vela/op"
        )

        // get the passed parameter
        parameter: {
          name: string
        }

        // apply component and traits
        apply: op.#ApplyComponent & {
          component: parameter.name
        }
```

Actually, `ApplyComponent` is a collection of operators, then we'll show how to apply components and traits separately.

## Apply components and traits separately

> The 'ApplyComponent' have integrated the `Load` and `Apply` operator. In addition, components and traits need to be applied separately.

Reviewer: ```suggestion
The 'ApplyComponent' have integrated the `Load` and `Apply` actions. In the following, we will apply components and traits separately.
```

```yaml
apiVersion: core.oam.dev/v1beta1
kind: WorkflowStepDefinition
metadata:
  name: apply-express
  namespace: default
spec:
  schematic:
    cue:
      template: |
        import (
        	"vela/op"
        )

        // get the passed parameter
        parameter: {
          name: string
        }

        // load the component
        load: op.#Load & {
        	component: parameter.name
        }

        // apply the component, `.value.workload` means the actual component
        apply: op.#Apply & {
        	value: {
        		load.value.workload
        	}
        }

        // apply the trait, `.value.auxiliaries[0]` means the actual trait since we have only one trait here
        apply: op.#Apply & {
          value: {
            load.value.auxiliaries[0]
          }
        }
```

> If we have multiple traits, we can also loop to apply.

Reviewer: ```suggestion
If we have multiple traits, we can also apply one by one in a loop:
```
